frontend/src/libraries/date-locale.ts

```diff
@@ -0,0 +1,24 @@
+import { Language } from "./i18n";
+import { Locale as RawLocale } from "date-fns";
+import { DurationFormatter } from "./locales";
+
+export type Locale = RawLocale & { durationFormatter: DurationFormatter };
+
+export const getDateLocale = async (
+  language: Language | "cimode"
+): Promise<Locale> => {
+  switch (language) {
+    case "en":
+      return (await import("./locales/en")).locale;
+    case "ru":
+      return (await import("./locales/ru")).locale;
+    case "vi":
+      return (await import("./locales/vi")).locale;
+    case "zh-Hans":
+      return (await import("./locales/zh-Hans")).locale;
+    case "uk":
+      return (await import("./locales/uk")).locale;
+    case "cimode":
```

Comment:
This case is needed for test snapshots, if so maybe we should add some explaining comment here?